msal/src/test/java/com/microsoft/identity/client/robolectric/tests/AcquireTokenMockTest.java

```diff
@@ -0,0 +1,577 @@
+@Config(shadows = {ShadowStorageHelper.class, ShadowAuthority.class, ShadowHttpRequest.class, ShadowMsalUtils.class})
+public final class AcquireTokenMockTest {
+
+    private static final String[] SCOPES = {"user.read"};
+    private static final String AAD_MOCK_AUTHORITY = "https://test.authority/mock";
+
+    @Test
+    public void canAcquireToken() {
+        new AcquireTokenBaseTest() {
+
+            @Override
+            void makeAcquireTokenCall(final IPublicClientApplication publicClientApplication,
+                                      final Activity activity) {
+
+                String username = "fake@test.com";
+
+                AcquireTokenParameters parameters = new AcquireTokenParameters.Builder()
+                        .startAuthorizationFromActivity(activity)
+                        .withLoginHint(username)
+                        .withScopes(Arrays.asList(SCOPES))
+                        .fromAuthority(AAD_MOCK_AUTHORITY)
+                        .callback(new AuthenticationCallback() {
+                            @Override
+                            public void onSuccess(IAuthenticationResult authenticationResult) {
+                                Assert.assertTrue(!StringUtil.isEmpty(authenticationResult.getAccessToken()));
+                            }
+
+                            @Override
+                            public void onError(MsalException exception) {
+                                fail(exception.getMessage());
+                            }
+
+                            @Override
+                            public void onCancel() {
+                                fail("User cancelled flow");
+                            }
+                        })
+                        .build();
+
+                publicClientApplication.acquireToken(parameters);
+                RoboTestUtils.flushScheduler();
+            }
+
+        }.performTest();
+    }
+
+    @Test
+    public void acquireTokenFailsIfLoginHintNotProvidedForRopc() {
+        new AcquireTokenBaseTest() {
+
+            @Override
+            void makeAcquireTokenCall(final IPublicClientApplication publicClientApplication,
+                                      final Activity activity) {
+
+                AcquireTokenParameters parameters = new AcquireTokenParameters.Builder()
+                        .startAuthorizationFromActivity(activity)
+                        .withScopes(Arrays.asList(SCOPES))
+                        .fromAuthority(AAD_MOCK_AUTHORITY)
+                        .callback(new AuthenticationCallback() {
+                            @Override
+                            public void onSuccess(IAuthenticationResult authenticationResult) {
+                                fail("Unexpected Success");
+                            }
+
+                            @Override
+                            public void onError(MsalException exception) {
+                                Assert.assertTrue(true);
```

Comment:
nit: double check that we're getting an expected MsalException?

Reply:
Addressed